From david-b@pacbell.net  Sun Jan 25 12:56:13 2009
From: Swaminathan S <swami.iyer@ti.com>
Date: Sat, 24 Jan 2009 17:57:37 -0800
Subject: USB: musb cppi dma fix
To: linux-usb@vger.kernel.org
Cc: Greg KH <greg@kroah.com>, Felipe Balbi <felipebalbi@users.sourceforge.net>
Message-ID: <200901241757.37411.david-b@pacbell.net>
Content-Disposition: inline


From: Swaminathan S <swami.iyer@ti.com>

Initializes the actual_len field to 0 before every DMA transaction.

Signed-off-by: Swaminathan S <swami.iyer@ti.com>
Acked-by: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/cppi_dma.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/musb/cppi_dma.c
+++ b/drivers/usb/musb/cppi_dma.c
@@ -995,6 +995,7 @@ static int cppi_channel_program(struct d
 	cppi_ch->offset = 0;
 	cppi_ch->maxpacket = maxpacket;
 	cppi_ch->buf_len = len;
+	cppi_ch->channel.actual_len = 0;
 
 	/* TX channel? or RX? */
 	if (cppi_ch->transmit)
